### DISCUSSION TITLE
I can't use MySQL driver in MariaDB

### DISCUSSION THREAD

**Discussion by rgonzlz** [mikro-orm/mikro-orm#5584] [General]
### Describe the bug

I am upgrading an application from v5 to v6 and we currently use the MySQL driver on the MariaDB server (11.0.2) because the elastic-apm-node library does NOT work with the Maria driver.

In theory both MySQL and MariaDB are compatible, but when I run something, I get the following error:

![image](https://github.com/mikro-orm/mikro-orm/assets/7152452/a735bd27-b95b-4e70-98fe-b8d6bbd5cab0)

What is the reference here https://github.com/mikro-orm/mikro-orm/blob/master/packages/mariadb/src/MariaDbSchemaHelper.ts#L34

https://github.com/mikro-orm/mikro-orm/blob/master/packages/knex/src/dialects/mysql/MySqlSchemaHelper.ts#L63

What can I do, is there a workaround available? 

Thank you.

### Reproduction

Use MysqlDriver over MariaDB

### What driver are you using?

@mikro-orm/mariadb

### MikroORM version

6.2.6

### Node.js version

20

### Operating system

macOS 14.4.1

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
I am not interested in cross supporting, there are differences on schema level.

**Reply by rgonzlz** [reply]
Thank you for your response.

Could you tell me where I could put the part about registering queries? I have seen that MikroORM has hooks or use the logger system (which would be the worst option), but I don't know where it could be placed to correctly intercept the executed queries.
